\documentclass[]{article}
\usepackage{setspace}
\usepackage[margin = 1in]{geometry}
\usepackage{todonotes}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={Chapter 1},
            pdfauthor={Frederick Boehm},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}
\usepackage[
    backend=biber,
    style=authoryear,
    natbib=true,
    url=true, 
    doi=true,
    eprint=true
]{biblatex}
\addbibresource{ch1.bib}
%\addbibresource{research.bib}
\addbibresource{../diss-latex/research.bib}

% Create subtitle command for use in maketitle
\newcommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{Chapter 1}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Frederick Boehm}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{\today}


\begin{document}
\doublespacing
\maketitle

\listoftodos
\listoffigures
\listoftables




Identification of genes that affect measurable traits has a long and
successful history in model organisms. Complex traits include
clinical measurements such as systolic blood pressure and body weight as
well as newly attainable biomolecular traits like gene expression
levels, protein concentrations, and lipid levels. Understanding the
genetic underpinnings of such traits may inform biology,
medicine, and public health.

We begin by discussing quantitative trait locus (QTL) mapping in two-parent crosses. We then discuss pleiotropy testing in two-parent crosses before considering newly developed multiparental populations. We discuss the design for one multiparental population, the Diversity Outbred mice. We then justify the need for a pleiotropy test for use in multiparental populations.


\subsection{QTL mapping in two-parent crosses}

QTL mapping is a systematic, statistical approach to identifying genetic
loci where genetic variation affects phenotypic variation in a measured trait.
We use the term ``locus'' to refer to a small, contiguous genomic region, often
several megabases in length. Standard inputs are genome-wide marker genotypes for a collection of study subjects and a set of trait measurements on the same subjects.

Two major statistical challenges are what \citet{broman2009guide} call 
the ``missing data'' problem and the ``model selection'' problem. 
The ``missing data'' problem arises in QTL studies because genotypes are obtained
at only select markers. In this sense, genotypes at positions between
markers are ``missing''.
A flexible framework for resolving the ``missing data'' problem involves a
hidden Markov model \citep{broman2009guide}. 
In this hidden Markov model, marker genotypes are observed random variables, 
while genotypes at intervening bases are unobserved random variables \citep{broman2009guide,broman2006use}. 
Two sets of recursive equations, termed ``forward'' and ``backward'' equations, enable efficient calculation of genotype probabilities (conditional on the observed marker genotypes) \citep{baum1970maximization}.


%Modern QTL studies have their origins in the work of \citet{sax1923association}. \citet{sax1923association}, working 30 years before \citet{watson1953molecular} reported the molecular structure of DNA, studied seed color and weight in the common bean (\emph{Phaseolus vulgaris}). He used seed color to partition beans into genetic marker classes

% QTL scan (univariate)
A univariate QTL scan is a procedure to interrogate the entire genome for genetic
variants that affect a specified trait of interest.
It requires as inputs genome-wide marker genotypes (or, equivalently,
genotype probabilities) and trait measurements for a collection of subjects.
One specifies the form of a statistical model and calculates the
likelihood of the model parameters given the observed (trait and marker) data. 
One calculates the likelihood at every marker.
Next, likelihood ratio test statistics are calculated for every marker.
These compare, for every marker, the null hypothesis that there is no 
QTL against the alternative that there is a QTL (at the specified marker).
In other words, one performs a likelihood ratio test for every marker across the genome.
One then uses a permutation test to determine a genome-wide critical value
for the likelihood ratio test statistic \citep{churchill1994empirical}. 

% relationship between peak position and QTL 2-lod interval


% model form - linear regression


% model form - random effects incorporation









% QTL scan (multivariate)

In anticipation that multivariate mapping of correlated traits would enhance statistical power to detect QTL and would improve precision of QTL positions, both \citet{jiang1995multiple} and \citet{korol1995interval} developed multivariate interval mapping procedures. 


In the context of multivariate QTL mapping, \citet{jiang1995multiple} 
proposed and developed a statistical hypothesis test for distinguishing
a single pleiotropic locus that affects multiple traits from two (or more)
separate QTL.
Their test applies to the setting in which multiple traits
map to a single genomic region \citep{jiang1995multiple}. 
That region, then, can be interrogated further by asking whether it
harbors a single pleiotropic QTL or multiple separate QTL. For simplicity, we focus on the case where two traits both map to a single genomic region, and we wish to determine whether the two traits share a single QTL or have two separate QTL. 











\subsection{Diversity Outbred Mice}

Geneticists sought a mammalian gene mapping resource 
that could be used for study of a wide variety of quantitative traits. 
The magnitude of such an undertaking required a collaborative, community-supported approach \citep{de2014genetics}. Scientists conceived of the Diversity Outbred (DO) mouse population as such a high-resolution gene mapping resource. They elected to seed the population with partially inbred progenitors of the Collaborative Cross mouse population. The Collaborative Cross mouse mating design involved starting with mice from eight inbred lines: A/J, C57BL/6J, 129S1/SvImJ, NOD/ShiLtJ, NZO/HILtJ, CAST/EiJ, PWK/PhJ, WSB/EiJ.

The designers used a ``funnel'' mating scheme to generate mice with DNA from all eight founder lines over the course of 3 generations. For example, in one funnel, mating pairs are: A x B, C x D, E x F, and G x H in the first generation. AB offspring would then mate with CD offspring and EF offspring would mate with GH mice. Finally, the ABCD mice would mate with the EFGH mice to create a generation of mice that contain genetic material from all eight inbred founder lines. Subsequent generations of inbreeding resulted in multiple inbred lines for the Collaborative Cross. \todo[inline]{make this figure}

The designers of the Diversity Outbred population started with \todo[inline]{get number of mice and from which generations}

Due to the breeding scheme, each Diversity Outbred mouse is a highly heterozygous mosaic of 
founders' DNA (Figure~)\todo{add figure with mosaics over multiple generations here}. 
With each generation, the Diversity Outbred mice accumulate meiotic recombinations. 
This is because each mouse inherits DNA from its parents, and each meiosis provides opportunities for recombinations. \todo[inline]{need to explain this better. How does Andrew explain it in his thesis?}

The cumulative effect of recombinations over many generations of outbreeding is to create mice whose DNA mosaic contains smaller and smaller contiguous pieces from each founder line. 

The Diversity Outbred mice enable high-resolution QTL mapping \citep{gatti2014quantitative}. One factor that contributes to high-resolution mapping is the ability to infer the founder line from which each marker's DNA arose. 




\subsection{QTL mapping in DO mice}

QTL mapping in Diversity Outbred mice, like that in mice from two-parent crosses, is a multi-step procedure: 1. data acquisition, 2. inference of missing genotypes, and 3. modeling phenotypes as a function of genotypes. Data acquisition involves measurement of phenotypes and, at specified genetic markers, termed ``single nucleotide polymorphism'' or SNP markers, measurement of two-allele genotypes. Often, the SNP marker genotypes are obtained by use of a microarray, such as the GigaMUGA SNP microarray \citep{morgan2015mouse}. 

The next step, missing genotypes inference, is needed because of the ``missing data problem'' \citep{broman2009guide}. It takes as input the two-allele genotypes at the measured SNP markers. An expectation maximization algorithm \citep{dempster1977maximum} for a hidden Markov model, developed by \citet{broman2012haplotype} and \citet{broman2012genotype} and implemented in the \texttt{qtl2} R package \citep{broman2019rqtl2}, outputs 36-state genotype probabilities for all nuclear autosomal markers and pseudomarkers. Pseudomarkers, as we use the term, are arbitrary nucleotide bases at which the researcher wants 36-state genotype probabilities. Finally, we collapse the 36-state genotype probabilities to eight founder allele dosages at each marker and for every mouse before performing linear regression to identify QTL for every univariate trait. 



\subsection{Testing pleiotropy in two-parent crosses (Jiang \& Zeng}

\citet{jiang1995multiple} developed a suite of methods for analyzing multivariate phenotypes in two-parent crosses. Among the novel methods in their article is a test of pleiotropy vs separate QTL. They explain that such a test is useful when two traits map to a single genomic region. The question then arises ``do the two traits associate with the same locus, or do they associate with separate loci''?

If both traits associate with the same locus, then that locus is called ``pleiotropic''. If the two traits associate with separate loci, then we say that they have separate QTL (one for each trait). While ``pleiotropy'' has multiple related usages in the genetics literature, we use ``pleiotropy'' here to refer to the situation in which one genetic locus affects two or more traits. 

In the development of their test of pleiotropy v separate QTL, Jiang and Zeng model the multivariate phenotype as the sum of a linear function of the genotypes and a random error term. They assume that the phenotypes matrix is related to the genotypes data (for a single marker) through the following equation:


\begin{equation}
Y = XB + E
\end{equation}


\noindent Where Y is a n by 2 matrix of phenotype values (with each row being one subject and each column being one trait), X is a n by 1 matrix of genotypes and B is a 1 by 2 matrix of allele effects. The random error is assumed to follow a normal distribution with mean zero and a positive variance. Jiang and Zeng (1995) also incorporate additional terms to model dominant effects and to control for residual genetic variation. We forego inclusion of dominant terms and additional markers for controlling residual genetic variation. 


\citet{jiang1995multiple} use a mixture model to relate genotypes to phenotypes. They observe that the there are 9 possible ordered pairs of genotypes for two markers in a F2 population. They thus approach the problem as one with a 9-state mixture model. They provide the equations needed for an expectation-conditional maximization (ECM) algorithm for fitting the mixture model. They use a chi-square distribution with one degree of freedom as the null distribution of the test statistic. 

Further investigations by \citet{jiang1995multiple} with simulations revealed reasonable behavior of the hypothesis test under the specified conditions. 
They learned that dense marker coverage tends to aid in distinguishing pleiotropy from separate QTL. \todo[inline]{look back at JZ 1995's simulations. How did they examine power and type I error rates? Do they examine univariate LOD strength when considering power? What about interlocus distance?}










 Finding values that minimize the sum of squared residuals - THINK ABOUT WHAT A RESIDUAL IS HERE… a matrix? - They use a computational algorithm called “expectation - conditional maximization” to find these values. ECM is an iterative algorithm. The user inputs starting values for the parameters and the algorithm changes them one at a time until the change in log likelihood between successive iterations is sufficiently small. The algorithm is then said to have ``converged''. The final iteration’s parameter values are treated as those that maximize the likelihood.


Jiang \& Zeng (1995), in the context of their multivariate QTL mapping methods, describe a test of pleiotropy vs separate QTL. They formulate their question as a statistical hypothesis test in which the null hypothesis is pleiotropy while the the alternative hypothesis is separate QTL. In other words, they consider the case in which the two traits map to the same locus (ie, pleiotropy) as a special case of the more general setting (separate QTL) in which the two traits may or may not map to the same locus. In statistical terminology, one would say that the parameter space is restricted under the null hypothesis.

Jiang and Zeng then calculate the likelihoods of the two models (that under the hypothesis of pleiotropy and that under the hypothesis of separate QTL) at each ordered pair of putative loci in the genomic region of interest. The likelihood ratio test statistic is the logarithm of the ratio of the maximum of the likelihoods under pleiotropy to the maximum of the likelihoods under the separate QTL hypothesis. 

Jiang and Zeng determined p-values for their test statistics by comparing them to a theoretical null distribution. In this specific case, Jiang and Zeng reasoned that the null distribution of their test statistic is a chi-squared distribution with 1 degree of freedom. Their justification for this is based on asymptotic statistics theory.

Schadt et al (2005) extended the methods of Jiang and Zeng (1995). 





\printbibliography


\end{document}
